modules/custom/moj_resources/src/Utilities.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Drupal\moj_resources;
+
+use Drupal\Core\Entity\Query\QueryInterface;
+
+/**
+ * Utility class for Prisoner Content Hub endpoints
+ *
+*/
+class Utilities {
+  /**
+    * Add filter by prison categories to a query object
+    *
+    * @param int $prisonId
+    * @param int[] $prisonCategories
+    * @param QueryInterface
+    *
+    * @return QueryInterface
+  */
+  public static function filterByPrisonCategories($prisonId, $prisonCategories, $query) {
+    $filterByPrisonId = $query
+      ->orConditionGroup()
+      ->condition('field_moj_prisons', $prisonId, '=')
+      ->notExists('field_moj_prisons');
+
+    $filterByPrisonCategories = $query
+      ->orConditionGroup()
+      ->condition('field_prison_categories', $prisonCategories, 'IN')
+      ->notExists('field_prison_categories');
+
+    return $query
```

Comment:
Need to ensure content owners tick the relevant prison categories that match the prison or none here if they want to filter just by prison. And the other way around if just selecting by prison category but logic looks good 👍

Reply:
Yup, agreed, being a utility function this is something we can iterate on across the board if needs be